src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -310,4 +344,16 @@ namespace hit {
     void HomomorphicEval::relinearize_inplace_internal(CKKSCiphertext &ct) {
         relinearize(get_evaluator(), ct.backend_ct, ct.backend_ct);
     }
+
+    CKKSCiphertext HomomorphicEval::bootstrap_internal(const CKKSCiphertext &ct, bool rescale_for_bootstrapping) {
+        if (rescale_for_bootstrapping && ct.he_level() == 0) {
+            LOG_AND_THROW_STREAM("Unable to bootstrap ciphertext at level 0 when rescale_for_bootstrapping is true.");
+        }
```

Comment:
Nit: may need to add some explanation why `level 0` and `rescale_for_bootstrapping` conflict. Or some comments from previous PR tells the relationship?

Reply:
Added comments here, but also see the API comment in evaluator.h:
> /* Refresh a ciphertext at a low level (e.g., 0) to a ciphertext at a higher level. This
         * enables the cryptosystem to continue to do work on a ciphertext. The exact level of the
         * output is determined by the cryptosystem parameters. Lattigo has a restriction on the
         * scale of the input ciphertext, namely, that if `ct.he_level() = 0`, then `ct.scale()`
         * must be an exact power of two smaller or equal to round(Q0/2^{10}). If you know this
         * condition is satisfied in your circuit, you can bootstrap directly, thereby allowing
         * you to use the last ciphertext level for computation. If you are not sure,
         * or if the ciphertext scale does not meet this requirement, the bootstrapping process can
         * implicitly consume one extra level to rescale your ciphertext appropriately. If you know
         * that rescaling is unnecessary, set `rescale_for_bootstrapping` to `false`, otherwise use
         * the default value of `true`.
         */